src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -292,6 +295,32 @@ public static void EndInvokeJS(JSRuntime jsRuntime, string arguments)
             }
         }
 
+        /// <summary>
+        /// Accepts the byte array data being transferred from JS to DotNet.
+        /// </summary>
+        /// <param name="jsRuntime">The <see cref="JSRuntime"/>.</param>
+        /// <param name="id">Identifier for the byte array being transfered.</param>
+        /// <param name="data">Byte array to be transfered from JS.</param>
+        public static void SupplyByteArray(JSRuntime jsRuntime, long id, byte[] data)
+        {
+            if (id == 0)
+            {
+                // Starting a new transfer, clear out previously stored byte arrays
+                jsRuntime.ResetByteArraysToBeRevived();
+            }
+            else if (id != (jsRuntime.ByteArraysToBeRevived.Count + 1))
+            {
+                throw new ArgumentOutOfRangeException($"Element id '{id}' cannot be added to the byte arrays to be revived with length '{jsRuntime.ByteArraysToBeRevived.Count}'.");
+            }
+            else if ((31 * 1024) - data.Length < jsRuntime.ByteArraysToBeRevivedByteLength) // TODO; get this limit from SignalR somehow?
```

Comment:
If you do want to get the message size limit from SignalR, you can add a DI dependency to `RemoteJSRuntime`'s constructor so it receives a `Microsoft.Extensions.Options.IOptions<Microsoft.AspNetCore.SignalR.HubOptions> HubOptions`, and then read the `.Value.MaximumReceiveMessageSize` from that.

I know it's a bit awkward to get the value into `DotNetDispatcher` because of the layering (it doesn't know about `RemoteJSRuntime`). You could declare some `internal protected virtual` property on `JSRuntime` that defines a `MaximumByteDataPerCall` with default value `int.MaxValue`, overridden in `RemoteJSRuntime`. Or I'm sure you can come up with other ways of doing it.